Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -594,16 +599,40 @@ void FunctionDefinitionWriter::visit(AST::Statement& statement)
 
 void FunctionDefinitionWriter::visit(AST::AssignmentStatement& assignment)
 {
-    m_stringBuilder.append(m_indent);
     visit(assignment.lhs());
     m_stringBuilder.append(" = ");
     visit(assignment.rhs());
     m_stringBuilder.append(";\n");
 }
 
+void FunctionDefinitionWriter::visit(AST::CompoundStatement& statement)
+{
+    m_stringBuilder.append("{\n");
+    {
+        IndentationScope scope(m_indent);
```

Comment:
IIUC you removed all the indentation from statements and added it here, I don't quite remember, can the other statements only appear inside a compound statement?

Reply:
My understanding is that all statements are enclosed by a compound statement, the top-most being the body of a function. I'll check with the spec ensure that's right.

Reply:
Getting the indentation right with nested-if is tricky...